src/main/java/bio/terra/cli/apps/DockerAppsRunner.java

```diff
@@ -124,12 +122,33 @@ public void runToolCommand(
     }
     envVars.putAll(terraReferences);
 
-    buildDockerClient();
+    // mount the global context directory and the workspace directory to the container
+    //  e.g. global context dir on host = $HOME/.terra
+    //       global context dir on container = CONTAINER_HOME_DIR/.terra
+    //       workspace dir on host = $HOME/terra-workspaces/workspace123
+    //       workspace dir on container = CONTAINER_WORKSPACE_DIR
+    Path globalContextDirOnContainer =
+        GlobalContext.getGlobalContextDir(Path.of(CONTAINER_HOME_DIR));
```

Comment:
I'm a little confused by the paths. We use `getGlobalContextDir()` for the host global context dir, but `getGlobalContextDir(Path relativeToDir)` to get the right container path? Similarly for `getPetSaKeyFile`'s two overloads? Should the relativeToDir overloads just be `getGlobalContextDirOnContainer`?
It feels like the path logic right now is split between this class and its `CONTAINER_` vars and the `GlobalContext`.

Reply:
spitballing suggestions, maybe relegate all the file logic to a single class? And maybe there's an instance of the class for host & one for the container? Right now there's some file path logic in both the GlobalContext & WorkspaceContext. I don't know if it makes sense to take file path resolution out of there.

Reply:
+1 on being confused here. I added a comment on the GlobalContext method itself, but IMO removing the "relativeTo" parameter and using explicit `getFooOnContainer` and `getFooOnHost` variants, as Will suggested, should help a lot.

Reply:
Yeah, this is the part I really struggled with. I wanted to keep the structure of the global and workspace context directories in those classes, but ran into trouble when trying to map that to the container paths.

I'm hesitant to split this class in two because almost all the code just uses the host machine version of the global context. So it feels like the container paths are a special case. I'll take a stab at moving the container-specific logic into the DockerAppsRunner class.

Reply:
I took another pass at this. PTAL to see if this seems more/less logical.
- Left all logic about the structure of the context directories in the Global/WorkspaceContext classes.
- Added helper methods to the DockerAppsRunner class to convert paths on the host machine to paths on the container.